src/FreqWindow.cpp

```diff
@@ -581,63 +611,35 @@ bool FrequencyPlotDialog::Show(bool show)
 
 void FrequencyPlotDialog::GetAudio()
 {
-   mData.reset();
-   mDataLen = 0;
+   mChannelGroupSampleViews.clear();
 
    int selcount = 0;
-   bool warning = false;
    for (auto track :
       TrackList::Get(*mProject).Selected<const WaveTrack>()
    ) {
       auto &selectedRegion = ViewInfo::Get(*mProject).selectedRegion;
       auto start = track->TimeToLongSamples(selectedRegion.t0());
-      if (selcount == 0) {
+      if (selcount == 0)
          mRate = track->GetRate();
-         auto end = track->TimeToLongSamples(selectedRegion.t1());
-         auto dataLen = end - start;
-         // Permit approximately 46.60 minutes of selected samples at
-         // a sampling frequency of 48 kHz (11.65 minutes at 192 kHz).
-         auto maxDataLen = size_t(2) << 26;
-         if (dataLen > maxDataLen) {
-            warning = true;
-            mDataLen = maxDataLen;
-         }
-         else
-            mDataLen = dataLen.as_size_t();
-         mData = Floats{ mDataLen };
-      }
       const auto nChannels = track->NChannels();
       if (track->GetRate() != mRate) {
          AudacityMessageBox(
             XO(
 "To plot the spectrum, all selected tracks must be the same sample rate.") );
-         mData.reset();
-         mDataLen = 0;
+         mChannelGroupSampleViews.clear();
          return;
       }
-      Floats buffer1{ mDataLen };
-      Floats buffer2{ mDataLen };
-      float *const buffers[]{ buffer1.get(), buffer2.get() };
       // Don't allow throw for bad reads
-      track->GetFloats(0, nChannels, buffers, start, mDataLen,
-         false, FillFormat::fillZero, false);
-      for (size_t i = 0; i < mDataLen; i++)
-         mData[i] = buffers[0][i];
-      for (size_t iChannel = 1; iChannel < nChannels; ++iChannel) {
-         const auto buffer = buffers[iChannel];
-         for (size_t i = 0; i < mDataLen; i++)
-            mData[i] += buffer[i];
-      }
+      constexpr auto mayThrow = false;
+      mChannelGroupSampleViews.push_back(track->GetSampleView(
+         selectedRegion.t0(), selectedRegion.t1(), mayThrow));
       ++selcount;
    }
 
-   if (selcount == 0)
-      return;
```

Comment:
That should not be deleted I think

Reply:
I don't think so. In either versions, before or after, the only thing it does is returning before the warning check that there aren't too many samples. But if `selcount == 0`, it means that there was no selected track in the project, and no sample was read, in which case, of course, no warning anyways.